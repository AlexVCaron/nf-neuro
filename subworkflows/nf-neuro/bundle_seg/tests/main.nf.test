nextflow_workflow {

    name "Test Subworkflow BUNDLE_SEG"
    script "../main.nf"
    workflow "BUNDLE_SEG"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_nfneuro"
    tag "subworkflows/bundle_seg"

    tag "bundle/recognize"
    tag "registration/ants"

    tag "load_test_data"

    tag "stub"
    options "-stub-run"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "tracking.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("rbx - base - no_atlas_folder") {

        when {
            workflow {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/fa.nii.gz")
                    ]
                }
                input[1] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/pft.trk")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out
                        .findAll{ channel -> !channel.key.isInteger() && channel.value  }
                        .collectEntries{ channel ->
                            [(channel.key): ["versions"].contains(channel.key)
                                ? channel.value
                                : channel.value.collect{ subject ->
                                    [ subject[0] ] + subject[1..-1].flatten().collect{ entry -> entry ? file(entry).name : "" }
                            } ]
                        }
                ).match() }
            )
        }
    }
}
