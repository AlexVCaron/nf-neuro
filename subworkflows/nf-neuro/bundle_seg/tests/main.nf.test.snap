{
    "rbx - base - no_atlas_folder": {
        "content": [
            {
                "bundles": [
                    [
                        {
                            "id": "test"
                        },
                        "test__AF_L_cleaned.trk"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2fbc09a51203e509d1a15a631168680e",
                    "versions.yml:md5,84a499436fbf8f1367ec4d8b21cf3abd"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-04T20:55:58.757696313"
    }
}